### PR TITLE
docs: create labeling guide

### LINKED ISSUE #7607: docs: create a tagging guide
**What would you like Renovate to be able to do?**

Document how to tag issues.
I recently got triage rights, and so can label issues.
I don't know about the back-end structure of Renovate, and so some labels are not very clear for me.
This makes me hesitant to tag away on the repository.
It also leads to questions fired at the maintainers whenever things are unclear.

We can prevent this with a proper "tagging guide".

**Did you already have any implementation ideas?**

Create a new file called `how_to_tag_issues.md` or something like that and put it in the directory that we use for other `development` type docs.

This file will describe what different categories of tags there are, and how they should be used.
Maybe we can also add a bit of background behind each category of tags, so that it's easier to learn how to use them.

Once we have figured out how to use each tag, we can also improve the descriptions that are in the GitHub label section.

**Are there any workarounds or alternative ideas you've tried to avoid needing this feature?**

Learn how to tag by trial and error, and ask the maintainer to help each time I'm not sure.

**Is this a feature you'd be interested in implementing yourself?**

Yes, I will need help from the maintainers with explaining each tag category, as I'm not familiar with the Renovate back-end code. :smile:

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Starting point for discussion how to label an issue properly.

## Context:

Closes #7607 

Please take a look at the `QUESTION:` markers that I've left in the Markdown file to see what parts I don't understand.

The main problem is that I haven't found a simple plain language explanation of what the current Renovate back-end is built out of in terms of components. I don't understand what `datasources`, `managers`, and `workers` are.

I myself would benefit from a short and simple basic introduction to the Renovate back-end basics, that shows where to find these definitions and files. This would also be helpful for on-boarding new outside contributors or team members.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] Unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
datasource / manager ... are currently the folder under lib. so label `manager:nuget` means related to `lib/manager/nuget`, platform / versioning ... are corresponding. see also the modules section on https://docs.renovatebot.com/

**Comment by viceice** [comment]
`worker` labels are a little be more complicated, as I've added them recently to differenciate bewteen the different internal renovate working stages. They are a lot more complex and have a lot cross references.

**Comment by HonkingGoose** [comment]
Thanks @viceice for the links to the relevant parts in the Renovate docs... I should have seen those while I was doing my refactoring, but I was focused on the linebreaks, not on the text.... :smile: 

I'll read those docs first, and make some changes to this pull request after I've understood those modules better. :smile:

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:23]: I removed `major` now and I'm not sure that `breaking` then needs its own category. It can maybe be related to feature/fix/refactor ?

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md]: Breaking is added if we determine it's a backwards-incompatible change that requires us to publish a new major release.

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:22]: Remove blocker, combine with critical instead

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:35]: This is an important field so we could start by copying but best if we expand it.

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:33]: Any collaborator in the repo with rights can assign any label

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:59]: Agreed, remove it

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:57]:
```suggestion
Use the `breaking` label for Issues or PRs which contain changes that are not backwards compatible and require a major release bump.
```

File: docs/development/issue_labeling.md
Comment (HonkingGoose) [reply docs/development/issue_labeling.md]: I've removed the `major` label, and moved the breaking label to the Type of issue section.
I'm not sure what `breaking` denotes though, see the new file for questions. :smile:

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:73]:
```suggestion
Add a label `good first issue` to issues that are small, easy to fix, and do-able for a newcomer.
This label is sometimes picked up by tools or websites that try to encourage people to contribute to open source.
```

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:78]: I use this to indicate - usually to the issue creator - that I need them or someone else to do some work here or it is unlikely to get done.

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:75]:
```suggestion
Add a label `needs reproduction` if nobody's reproduced it in a public repo yet and such a reproduction is necessary before further work can be done.
```

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:89]:
```suggestion
```

These can all be removed

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:98]: Any of the platforms can be self-hosted, while both GitHub and GitLab can be hosted. I use this label to identify when an issue is applicable only to users who self-administer their own bot.

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:119]: When it is applicable specifically to particular datasources (e.g. as defined in the docs list of datasources)

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:193]: No, because it could be self-hosted *and* any of the above.

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:207]: Worker is kind of the "core" logic of Renovate. I think it's a good idea to have these, or they haven't been used heavily so far. In theory any feature or bug should apply to either a platform, manager, datasource, versioning or a worker part.

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:221]: I don't know either. Also, I would be fine to stop using `released` as a label because semantic-release already comments in every issue and PR anyway.

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/development/issue_labeling.md
Comment (HonkingGoose) [reply docs/development/issue_labeling.md:33]: I know that any collaborator with the required rights can assign labels, the question I'm really asking here is: "Who decides what priority an issue is? Is a best-effort guess from any collaborator okay?"

I hope I'm making sense here... :smile:

File: docs/development/issue_labeling.md
Comment (HonkingGoose) [reply docs/development/issue_labeling.md:215]: I think you'll need to edit some `semantic` bot configuration to get it to stop using the label?

File: docs/development/issue_labeling.md
Comment (rarkins) [reply docs/development/issue_labeling.md:24]: I've removed it from the repo

File: docs/development/issue_labeling.md
Comment (rarkins) [reply docs/development/issue_labeling.md:37]: I've added descriptions to those which were missing

File: docs/development/issue_labeling.md
Comment (rarkins) [reply docs/development/issue_labeling.md:35]: Yes, that is ok. Nothing bad happens if someone has a different opinion on priority.

File: docs/development/issue_labeling.md
Comment (rarkins) [reply docs/development/issue_labeling.md:209]: See https://github.com/renovatebot/renovate/pull/7654

Comment (HonkingGoose) [reply docs/development/issue_labeling.md]: I've removed the entire section now that you've merged #7654.

**Comment by HonkingGoose** [comment]
I noticed the labels names have changed. I like the new labels, they're very clear.

I'll go update this pull request again with the latest labels. I'm marking this PR as a draft until I've made those changes.

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:16]:
```suggestion
- To learn more about versioning, read the [Renovate docs on Versioning](https://docs.renovatebot.com/modules/versioning/).

Most issues should have a label relating to either a platform, manager, datasource, versioning or worker topic.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Comment by HonkingGoose** [comment]
@viceice asked me if we have a label for `nuget` versioning:
> do we have a label for nuget versioning?

And @viceice  also commented:

>No Problem, I think we need versioning labels too. Here the problem is nuget versioning. Maybe the nuget manager needs a special handling too. :thinking:

See #7659 to see those comments in their full context.

Currently we don't have any labels like `versioning:<ecosystem>`, I'll let @rarkins and team decide on whether or not we need more labels.

**Comment by rarkins** [comment]
We can add them as we need them

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:24]:
```suggestion
```

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:36]:
```suggestion
Add the `breaking` label for Issues or PRs which contain changes that are not backwards compatible and require a major version bump.
```

File: docs/development/issue_labeling.md
Comment (rarkins) [inline docs/development/issue_labeling.md:52]:
```suggestion
Nothing bad will happen if you select a "wrong" priority.
At a high level: critical = needs immediate fix, important = to be prioritized ahead of others, normal = default priority, low = trivial issue, or impacts a very small % of the user base.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Comment by rarkins** [comment]
Ready to merge after these requested changes

**Comment by HonkingGoose** [comment]
Jay! It's merged! :partying_face: :tada: 

Thank you @viceice and @rarkins for the help with this. :+1:

**Pull request merged**
